trio/_core/_multierror.py

```diff
@@ -176,6 +176,13 @@ class MultiError(BaseException):
 
     """
 
+    def __init__(self, exceptions):
+        # Avoid recursion when exceptions[0] returned by __new__() happens
+        # to be a MultiError and subsequently __init__() is called.
+        if self in exceptions:
+            return
+        self.exceptions = exceptions
```

Comment:
Let's also point to #1066 here to explain why we even have an `__init__` method.

Also, how about making the check:

```python
if hasattr(self, "exceptions"):
    # __init__ was already called on this object
    assert len(exceptions) == 1 and exceptions[0] is self
    return
```

Reply:
done, though I explained the reason directly rather than reference a historical bug